src/reporting_function.tf

```diff
@@ -0,0 +1,139 @@
+resource "azurerm_resource_group" "reporting_fdr_rg" {
+  count = var.reporting_fdr_enabled ? 1 : 0
+
+  name     = format("%s-reporting-fdr-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host reporting-fdr function
+module "reporting_fdr_function_snet" {
+  count                                          = var.reporting_fdr_enabled && var.cidr_subnet_reporting_fdr != null ? 1 : 0
+  source                                         = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v1.0.51"
+  name                                           = format("%s-reporting-fdr-snet", local.project)
+  address_prefixes                               = var.cidr_subnet_reporting_fdr
+  resource_group_name                            = azurerm_resource_group.rg_vnet.name
+  virtual_network_name                           = module.vnet.name
+  enforce_private_link_endpoint_network_policies = true
+
+  delegation = {
+    name = "default"
+    service_delegation = {
+      name    = "Microsoft.Web/serverFarms"
+      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
+    }
+  }
+}
+
+module "reporting_fdr_function" {
+  count  = var.reporting_fdr_enabled ? 1 : 0
```

Comment:
As per my comment above .... do we really really need this flag **reporting_fdr_enabled**.
Why not just create the function in all envs?

Reply:
see above response : )